src/drivers/mct/main/cime_comp_mod.F90

```diff
@@ -2094,6 +2095,10 @@ subroutine cime_run()
     logical            :: drv_pause  ! Driver writes pause restart file
     character(len=CL)  :: drv_resume ! Driver resets state from restart file
 
+    type(ESMF_Time)    :: etime_curr            ! Current model time
```

Comment:
I introduced local variables in this subroutine. I notice that, for the most part, this module uses module-level variables rather than subroutine-local variables. I couldn't bring myself to follow that convention, but could change this if you feel within-module consistency is more important than following typical Fortran coding practices.